pandas-stubs/core/frame.pyi

```diff
@@ -103,12 +105,14 @@ class DataFrame:
     def drop(self, index: Union[List[_str], Index], axis: _AxisType = ...) -> DataFrame: ...
     def drop_duplicates(self, keep: Union[_str, bool] = ...) -> DataFrame: ...
     def dropna(self, axis: int = ..., how: _str = ...) -> DataFrame: ...
+    def fillna(self, value: Union[Scalar, Dict, Series, DataFrame], method: str = None, axis: Union[str, int] = None, inplace: bool = False, limit: int = None, downcast: Dict = None) -> Union[DataFrame, None]: ...
```

Comment:
Same here; replace the default values by `...`.

There is another problem here: you are _returning_ a `Union`. This should be avoided if possible. And in this case it is possible to avoid it, with `@overload`:
```suggestion
    @overload
    def fillna(self, value: Union[Scalar, Dict, Series, DataFrame], method: str = ..., axis: Union[str, int] = ..., inplace: Literal[True], limit: int = ..., downcast: Dict = ...) -> None: ...
    @overload
    def fillna(self, value: Union[Scalar, Dict, Series, DataFrame], method: str = ..., axis: Union[str, int] = ..., inplace: Literal[False] = ..., limit: int = ..., downcast: Dict = ...) -> DataFrame: ...
```

The way it works is this: if you do not specify `inplace` or set it to `False` (`inplace=False`), then the second overload is used and a DataFrame is returned. If instead you set `inplace=True`, then the first overload is used and nothing is returned.